### DISCUSSION TITLE
Writing-Coding CC/SL 2021‘ Trafic failed

### DISCUSSION THREAD

**Discussion by DaZeuss** [cedricp/ddt4all#1147] [Q&A]
Hello dear experts, 
I have a serious problem with my 2021' Renault Trafic. I have retrofitted and tested all components for the originally cruise control / speed limiter. All buttons and switches work. I was able to check this with DDT4All. In the instrument cluster I was able to activate the cruise control / speed limiter, unfortunately this was not possible in the engine control unit. There, the value does not change after writing. 
I get the following message: 
Send ELM request: 2E 22 2F 01 
Bad Elm response: Wrong Response : No : Conditions Not Correct or Request Sequence Error (7F2E22)

I use the latest software from DDT4All, an OBDLinkSX (original from the manufacturer because fear of a fake adapter), Windows 11.

I retrofitted the cruise control / speed limiter in my previous 2019' Trafic in the same way. Everything worked very simply there, I am now at a loss and hope that you can help me.

![IMG_2888](https://github.com/cedricp/ddt4all/assets/173196799/2ea05c61-d60f-4da2-b1ed-5621d90471b6)
![IMG_2889](https://github.com/cedricp/ddt4all/assets/173196799/f28d58e5-a447-4919-9be4-098b6fa628cb)

**Comment by AndyIreland** [comment]
could it be maybe because is a 2021 model ... and as it is a 2021 model the OBD2 will surely be protected by a security gateway that needs unlocking before any parameters can be changed?

2019 car possibly had no security gateway on the obd2 CAN network

**Comment by DaZeuss** [comment]
Hi Andy,
I haven't found a security gateway in the vehicle so far. At least seems well hidden. How do you unlock the security gateway?

I wonder why I could write the values in the instrument cluster, but not in the engine control unit. Shouldn't the Security Gateway protect the entire system from my access?

**Reply by AndyIreland** [reply]
true - I would have thought that too that if it was security gateway blocked then you would not be able to program any ECU's 

Maybe then you are not meeting the requirements to program what you want to change. 
Try changing ms of can Timeout to 600ms
Try programming with engine turned off
Try programming with engine off but keycard in slot (ACC mode) 
Try programming with engine off and car in 1st gear (if car in manual) or 'D' if its an auto
Try programming after long press of Start button on dash with keycard in slot until 'please remove keycard shows on dash , but you are not actually starting the car so dont touch the brake or clutch (this is called park mode) 

To unlock gateway if you do have one on your car you have to purchase a token from renault services to open up gateway or some people reckon they have bypassed the gateway by taking apart their dashboard and wiring up OBD2 cable directly to their BCM unit. 
Good luck.

More reading about gateway here: 
https://github.com/cedricp/ddt4all/discussions/956
